Common/Utils/include/CommonUtils/BoostSerializer.h

```diff
@@ -31,63 +39,9 @@ namespace o2
 {
 namespace utils
 {
-namespace check
-{
-///A set of classes and struct to be sure the serialised object is either trivial or implementing custom serialize
-template <class Type, class Archive, typename = typename std::enable_if<std::is_class<Type>::value>::type>
-class is_boost_serializable
-{
- private:
-  struct TypeOverloader {
-    void serialize(Archive& ar, const unsigned int version) {}
-  };
-  struct TypeExt : public Type, public TypeOverloader {
-  };
-  template <typename T, T t>
-  class DeductionHelper
-  {
-  };
-
-  class True
-  {
-    char m;
-  };
-  class False
-  {
-    True m[2];
-  };
-
-  template <typename TestType>
-  static False deduce(TestType*, DeductionHelper<void (TypeOverloader::*)(), &TestType::serialize>* = 0);
-  static True deduce(...);
-
- public:
-  static const bool value = (sizeof(True) == sizeof(deduce((TypeExt*)(0))));
-};
-} // namespace check
-
 template <typename ContT>
-typename std::enable_if<check::is_boost_serializable<ContT, boost::archive::binary_oarchive>::value
-                        && std::is_class<typename ContT::value_type>::value, std::ostringstream>::type
-  SerializeContainer(const ContT& dataSet)
+std::ostringstream BoostSerialize(const ContT& dataSet)
 {
-  static_assert(check::is_boost_serializable<typename ContT::value_type, boost::archive::binary_oarchive>::value,
-                "This class doesn't provide a boost serializer.");
-  /// Serialises a container (vector, array or list) using boost serialisation routines.
-  /// Requires the contained type to be either trivial or provided with an overried of boost::serialise method.
-  std::ostringstream buffer;
-  boost::archive::binary_oarchive outputArchive(buffer);
-  outputArchive << dataSet;
-  return buffer;
-}
-
-template <typename ContT, typename ContentT = typename ContT::value_type>
-typename std::enable_if<check::is_boost_serializable<ContT, boost::archive::binary_oarchive>::value
-                        && !(std::is_class<ContentT>::value), std::ostringstream>::type
-  SerializeContainer(const ContT& dataSet)
-{
-  static_assert(boost::serialization::is_bitwise_serializable<typename ContT::value_type>::value,
-                "This type doesn't provide a boost serializer.");
   /// Serialises a container (vector, array or list) using boost serialisation routines.
   /// Requires the contained type to be either trivial or provided with an overried of boost::serialise method.
```

Comment:
I think this routine is now quite general, it does not need to be a container, can be a single element of a boost-serializable type as well.

Somehow I liked the conditional compilation depending on the type being boost serializable. The compilation error is probably a bit convoluted if the routine is used with a non-boost-serializable type

Reply:
Yes, I see the point. I'll think a bit about it :)

Reply:
more for the long run, nothing for this PR